src/main/fc/fc_core.c

```diff
@@ -206,11 +206,12 @@ void updateArmingStatus(void)
             unsetArmingDisabled(ARMING_DISABLED_CALIBRATING);
         }
 
-        if ((isModeActivationConditionPresent(BOXPREARM) && IS_RC_MODE_ACTIVE(BOXPREARM) && !ARMING_FLAG(WAS_ARMED_WITH_PREARM))
-            || !isModeActivationConditionPresent(BOXPREARM)) {
-            unsetArmingDisabled(ARMING_DISABLED_NOPREARM);
-        } else {
-            setArmingDisabled(ARMING_DISABLED_NOPREARM);
+        if (isModeActivationConditionPresent(BOXPREARM)) {
```

Comment:
It isn't, however it results in the same functionality.

if `isModeActivationConditionPresent(BOXPREARM)` never evaluates to true then `ARMING_DISABLED_NOPREARM` is never set, hence does not need to be unset.